File: src-tauri/src/models/conversation.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Conversation {
    pub id: String,
    pub title: String,
    pub selected_agent_ids: Vec<String>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateConversationInput {
    pub title: String,
    pub selected_agent_ids: Vec<String>,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UpdateConversationInput {
    pub id: String,
    pub title: Option<String>,
    pub selected_agent_ids: Option<Vec<String>>,
}

File: src/types/conversation.ts
export interface Conversation {
  id: string;
  title: string;
  selectedAgentIds: string[];
  createdAt: string;
  updatedAt: string;
}

export interface CreateConversationInput {
  title: string;
  selectedAgentIds: string[];
}

export interface UpdateConversationInput {
  id: string;
  title?: string;
  selectedAgentIds?: string[];
}
